packages/server/lib/saved_state.js

```diff
@@ -49,17 +50,23 @@ const formStatePath = (projectRoot) => {
 
     debug('missing project path, looking for project here')
 
-    const cypressJsonPath = cwd('cypress.json')
+    return CYPRESS_CONFIG_FILES.reduce((acc, filename) => {
+      return acc.then((projectRoot) => {
```

Comment:
I found this confusing - took me a while to figure out what was going on. Why not just use a for loop and synchronous operations? Less code, and it's a lot more simple (but still avoids mutation). Not tested, but something like this:

```js
    for (const filename of CYPRESS_CONFIG_FILES) {
      const cypressConfigPath = cwd(filename)
      const found = fs.pathExistsSync(cypressConfigPath)
      if (found) {
        const projectRoot = cwd()
        const fileName = 'state.json'
        return path.join(appData.toHashName(projectRoot), fileName)
      }
    }

    return path.join('__global__', fileName)
```

Could use reduce as well, but I think we want to return as soon as we find a config, so a for loop makes more sense.


Definitely easier to read without `async` and `then`.